File: AgoraChat-ObjectiveC/AgoraChat/Code/ReportMessage/ReportMessageItems/Cells/ACDReportMessageReasonCell.m
//
//  ACDReportMessageReasonCell.m
//  AgoraChat
//
//  Created by 杜洁鹏 on 2022/7/13.
//  Copyright © 2022 easemob. All rights reserved.
//

#import "ACDReportMessageReasonCell.h"

@interface ACDReportMessageReasonCell ()<UITextViewDelegate>

@end

@implementation ACDReportMessageReasonCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
    self.textView.delegate = self;
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];

    // Configure the view for the selected state
}

- (BOOL)textView:(UITextView *)textView shouldChangeTextInRange:(NSRange)range replacementText:(NSString *)text {
    if (textView.text.length + text.length > 500) {
        return NO;
    }

    return YES;
}

- (void)textViewDidChange:(UITextView *)textView {
    self.numLabel.text = [NSString stringWithFormat:@"%lu/500", (unsigned long)textView.text.length];
}

@end

File: AgoraChat-Swift/AgoraChat-Swift/code/ReportMessage/ReportMessageItems/Cells/ACDReportMessageReasonCell.swift
//
//  ACDReportMessageReasonCell.swift
//  AgoraChat-Swift
//
//  Created by 冯钊 on 2022/10/24.
//

import UIKit

class ACDReportMessageReasonCell: UITableViewCell {

    @IBOutlet weak var textView: UITextView!
    @IBOutlet private weak var numLabel: UILabel!
}

extension ACDReportMessageReasonCell: UITextViewDelegate {
    func textView(_ textView: UITextView, shouldChangeTextIn range: NSRange, replacementText text: String) -> Bool {
        return textView.text.count + text.count <= 500
    }
    
    func textViewDidChange(_ textView: UITextView) {
        self.numLabel.text = "\(self.textView.text.count)/500"
    }
}
